hack/update-helm-index.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash -e
+
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o pipefail
+
+asset_urls="$@"
+
+GH_REPO_PULL_URL="https://github.com/NVIDIA/k8s-device-plugin.git"
+git clone --local $$(pwd) $(HELM_REPO_PATH)
+git -C $(HELM_REPO_PATH) remote add github $(GH_REPO_PULL_URL)
+git -C $(HELM_REPO_PATH) fetch github gh-pages
+git -C $(HELM_REPO_PATH) checkout gh-pages
+
+pushd $HELM_REPO_PATH/stable > /dev/null
+# Download charts from release assets
+for asset_url in $asset_urls; do
+    if ! echo "$asset_url" | grep -q '.*tgz$'; then
+        echo "Skipping $asset_url, does not look like a Helm chart archive"
+        continue
+    fi
+
+    echo "Downloading $asset_url..."
+    curl -sSfLO $asset_url
+    # We rely on all release assets having the same baseurl
+    download_baseurl=`dirname $asset_url`
+done
+
+if [ -z "$download_baseurl" ]; then
+    echo "No Helm chart release assets found"
+    exit 0
+fi
+
+echo "Updating helm index"
+helm repo index stable --merge index.yaml --url https://nvidia.github.io/k8s-device-plugin/stable
+popd > /dev/null
+
+cp -f $HELM_REPO_PATH/stable/index.yaml $HELM_REPO_PATH/index.yaml
+
+pushd $HELM_REPO_PATH > /dev/null
+# Check if there were any changes in the repo
+if [ -z "`git status --short`" ]; then
+    echo "No changes in Helm repo index, gh-pages branch already up-to-date"
+    exit 0
+fi
+
+# Create a new commit
+commit_msg="Update Helm repo index for release $VERSION"
```

Comment:
We're doing more than this. We're also adding the actual packages.

Reply:
will enhance